runhouse/resources/packages/package.py

```diff
@@ -376,54 +383,39 @@ def to(
         path: Optional[str] = None,
     ):
         """Copy the package onto filesystem or cluster, and return the new Package object."""
-        from runhouse.resources.folders import Folder
-
-        if not isinstance(self.install_target, Folder):
+        if not isinstance(self.install_target, tuple):
             raise TypeError(
-                "`install_target` must be a Folder in order to copy the package to a system."
+                "`install_target` must be a Tuple in order to copy the package to a system."
             )
 
-        if (
-            isinstance(self.install_target.system, str)
-            and not self.install_target.system == "file"
-        ):
-            self.install_target.system = _get_cluster_from(self.install_target.system)
-
-        install_system_name = (
-            self.install_target.system.name
-            if isinstance(self.install_target.system, Cluster)
-            else self.install_target.system
-        )
         system = _get_cluster_from(system)
-        system_name = system.name if isinstance(system, Cluster) else system
-
-        if system_name == install_system_name:
+        if isinstance(system, Cluster) and system.on_this_cluster():
             return self
 
-        if isinstance(system, Resource):
-            if isinstance(self.install_target.system, Resource):
-                # if these are both clusters, check if they're pointing to the same address.
-                # We use endpoint instead of address here because if address is localhost, we need to port too
-                if self.install_target.system.endpoint(
-                    external=False
-                ) == system.endpoint(external=False):
-                    # If we're on the target system, just make sure the package is in the Python path
-                    sys.path.insert(0, self.install_target.local_path)
-                    return self
-
-            logger.info(
-                f"Copying package from {self.install_target.fsspec_url} to: {getattr(system, 'name', system)}"
-            )
         self._validate_folder_path()
+        if isinstance(system, Cluster):
+            local_path, path_to_sync_to_on_cluster = self.install_target
+            # Take the passed in path as priority
+            path_to_sync_to_on_cluster = (
+                path or path_to_sync_to_on_cluster or f"~/{Path(local_path).name}"
+            )
 
-        # sync folder to the fs and create a new module on the fs where relevant
-        new_folder = self.install_target.to(system, path=path)
-        new_folder = new_folder._destination_folder(
-            dest_path=new_folder.path, dest_system=system
-        )
-        new_package = copy.copy(self)
-        new_package.install_target = new_folder
-        return new_package
+            if not system._folder_exists(path_to_sync_to_on_cluster):
```

Comment:
Why do we only sync if the folder doesn't already exist? How do we propogate updates?